### ISSUE TITLE
Batched grad and gradgrad checks failing for empty input tensors

### ISSUE THREAD

**Issue by heitorschueroff** [pytorch/pytorch#53407]
## 🐛 Bug

I am getting the following error when running batched grad and gradgrad checks through OpInfo with empty input tensors

```
RuntimeError: While computing batched gradients, got: cannot reshape tensor of 0 elements into shape [-1, 0] because the unspecified dimension size -1 can be any value and is ambiguous
```

## Expected behavior

Batched grad and gradgrad checks should work with empty input tensors.

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @gqchen @pearu @nikitaved @soulitzer @mruberry @VitalyFedyunin @walterddr

**Comment by zou3519** [comment]
@heitorschueroff do you have an example OpInfo test (or PR) we can use to repro the problem?

**Comment by albanD** [comment]
High pri to investigate the exact cause of the issue.

**Comment by heitorschueroff** [comment]
@zou3519 Yes, see [here](https://github.com/pytorch/pytorch/pull/52859/files#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R1967).

**Comment by zou3519** [comment]
I dug into this a little. This isn't a generic batched gradient issue: batched gradient computation does work with empty tensors (i.e. tensors with zero-sized dimensions):
```
x = torch.randn(0, 5)
f = lambda x: x ** 2

jac = torch.autograd.functional.jacobian(f, x, vectorize=True)
```

The problem here is really that torch.matmul doesn't gracefully handle zero-sized dimensions:
```

**Comment by zou3519** [comment]
Speaking of which, I can't find a label for "tensor with zero-sized dimension support". We seem to have a couple issues like that (e.g. https://github.com/pytorch/pytorch/issues/50192)
